detection_rules/rule.py

```diff
@@ -462,6 +460,61 @@ class ThresholdCardinality:
     threshold: ThresholdMapping
 
 
+@dataclass(frozen=True)
+class NewTermsRuleData(QueryRuleData):
+    """Specific fields for new terms field rule."""
+
+    @dataclass(frozen=True)
+    class NewTermsMapping(MarshmallowDataclassMixin):
+        @dataclass(frozen=True)
+        class HistoryWindowStart:
+            field: definitions.NonEmptyStr
+            value: definitions.NonEmptyStr
+
+        field: definitions.NonEmptyStr
+        value: definitions.NewTermsFields
+        history_window_start: List[HistoryWindowStart]
+
+    type: Literal["new_terms"]
+    new_terms: NewTermsMapping
+
+    def validate(self, meta: RuleMeta) -> None:
+        """Validates terms in new_terms_fields are valid ECS schema."""
+
+        super(NewTermsRuleData, self).validate_query(meta)
+
+        # validate history window start field exists
+        assert self.new_terms.history_window_start, \
+            "new terms field found with no history_window_start field defined"
+
+        # ecs validation
+        stack_version = Version(meta.get("min_stack_version",
+                                Version(Version(load_current_package_version()) + (0,))))
```

Comment:
```suggestion
        min_stack_version = meta.get("min_stack_version")
        if min_stack_version is None:
            min_stack_version = str(Version(Version(load_current_package_version()) + (0,)))

        stack_version = Version(min_stack_version)
```

Reply:
This is a good point, remember that all rules at this time are min-stacked at 8.3 and if not, a unit test will fail because of build-time fields. I do agree that we should still account for no min-stacked provided as we have done before.